commit 16fa414aff02365534b7bbeab281c731b9c4497d
Author: Mark Wielaard <mjw@redhat.com>
Date:   Thu Aug 22 13:45:16 2013 +0200

    libdwfl/linux-kernel-modules.c (report_kernel): Pass add_p_vaddr as true.
    
    On some architectures (e.g. x86_64) the vmlinux is ET_EXEC, while on
    others (e.g. ppc64) it is ET_DYN.  In both cases the phdr p_vaddr will
    be non-zero.  We want the image to be placed as if it was ET_DYN, so
    pass true for add_p_vaddr which will do the right thing (in combination
    with a zero base) in either case.
    
    Signed-off-by: Mark Wielaard <mjw@redhat.com>

diff --git a/libdwfl/ChangeLog b/libdwfl/ChangeLog
index 88bdfc6..90ba686 100644
--- a/libdwfl/ChangeLog
+++ b/libdwfl/ChangeLog
@@ -1,3 +1,8 @@
+2013-08-25  Mark Wielaard  <mjw@redhat.com>
+
+	* linux-kernel-modules.c (report_kernel): Pass add_p_vaddr as true
+	to dwfl_report_elf.
+
 2013-07-25  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
 	* dwfl_segment_report_module.c (dwfl_segment_report_module): Check for
diff --git a/libdwfl/linux-kernel-modules.c b/libdwfl/linux-kernel-modules.c
index dec1a59..fe01028 100644
--- a/libdwfl/linux-kernel-modules.c
+++ b/libdwfl/linux-kernel-modules.c
@@ -216,8 +216,14 @@ report_kernel (Dwfl *dwfl, const char **release,
 
       if (report)
 	{
+	  /* Note that on some architectures (e.g. x86_64) the vmlinux
+	     is ET_EXEC, while on others (e.g. ppc64) it is ET_DYN.
+	     In both cases the phdr p_vaddr load address will be non-zero.
+	     We want the image to be placed as if it was ET_DYN, so
+	     pass true for add_p_vaddr which will do the right thing
+	     (in combination with a zero base) in either case.  */
 	  Dwfl_Module *mod = INTUSE(dwfl_report_elf) (dwfl, KERNEL_MODNAME,
-						      fname, fd, 0, false);
+						      fname, fd, 0, true);
 	  if (mod == NULL)
 	    result = -1;
 	  else
